core/trino-main/src/test/java/io/trino/sql/planner/assertions/PlanMatchPattern.java

```diff
@@ -632,28 +632,59 @@ public static PlanMatchPattern exchange(PlanMatchPattern... sources)
         return node(ExchangeNode.class, sources);
     }
 
+    public static PlanMatchPattern exchange(ExchangeNode.Scope scope, PlanMatchPattern... sources)
```

Comment:
just add one method that you require in test.
nit: Ideally, we have builder like: `io.trino.sql.planner.assertions.RowNumberMatcher.Builder`

Reply:
I removed not unnecessary methods.

> nit: Ideally, we have builder like: io.trino.sql.planner.assertions.RowNumberMatcher.Builder

IMO at this point, it's better to have static methods and introduce builder when there is more variability in the matcher usage.